firestore.rules

```diff
@@ -8,19 +8,16 @@ service cloud.firestore {
     match /users/{userId} {
        function validateAttrs(){
         return (
-      		  request.resource.data.keys().hasOnly(['displayName', 'avatar']) &&
+      	request.resource.data.keys().hasOnly(['displayName', 'avatar']) &&
         request.resource.data.displayName is string && request.resource.data.displayName.trim().size() >= 3 && request.resource.data.displayName.trim().size() <= 15 &&
-        // TODO: finish display name regex
-        // request.resource.data.displayName.match('/[0-9A-Za-z_-]/')
-        request.resource.data.avatar is number && request.resource.data.avatar >= 0 && request.resource.data.avatar <=12
+        request.resource.data.displayName.matches('[A-Za-z0-9-_]+') &&
+        request.resource.data.avatar is number &&  int(request.resource.data.avatar) == request.resource.data.avatar &&  request.resource.data.avatar >= 0 && request.resource.data.avatar <=12
```

Comment:
lets try to keep lines less than around 120 charicters long.

you can check by placing you're cusor at the end of the line and seeing that VS-Code says "Ln 14, Col 192" at the bottom. this means that you are on line 14 and coloumn 192 which means that the line is longer than 120 charicters